frontend/src/shape/BarStatShape.ts

```diff
@@ -0,0 +1,15 @@
+import { assertEqual, Shape } from "./Shape";
+import { StatMold, StatShape } from "./StatShape";
+
+export type BarStatShape = StatShape & {
+    type: "bar";
+};
+
+export class BarStatMold extends Shape<BarStatShape> {
```

Comment:
Minor concern about a Mold extending a Shape, I feel this would be clearer as an implementation, given that a Mold is not a type of Shape.